datafusion/catalog/src/table.rs

```diff
@@ -33,7 +33,19 @@ use datafusion_expr::{
 };
 use datafusion_physical_plan::ExecutionPlan;
 
-/// Source table
+/// A named table which can be queried.
+///
+/// Please see [`CatalogProvider`] for details of implementing a custom catalog.
```

Comment:
I think this link is important as this page provides the overall catalog API docs: https://docs.rs/datafusion/latest/datafusion/catalog/trait.CatalogProvider.html